flink-java/src/test/java/org/apache/flink/api/java/operator/GroupingTest.java

```diff
@@ -551,6 +551,14 @@ public Long getKey(Tuple4<Integer, Long, CustomType, Long[]> value) throws Excep
 						}, Order.ASCENDING);
 	}
 
+	@Test
+	public void testGroupingAtomicType() {
+		final ExecutionEnvironment env = ExecutionEnvironment.getExecutionEnvironment();
+		DataSet<Integer> dataSet = env.fromElements(0, 1, 1, 2, 0, 0);
+
+		dataSet.groupBy("*");
+	}
+
```

Comment:
Please also add tests that check for correct behavior if users define invalid programs (more than one expression, no wildcard, no key type).